internal/controller/controllers/cluster_controller.go

```diff
@@ -171,17 +228,121 @@ func (r *ClusterReconciler) updateStatus(ctx context.Context, cluster *adiiov1al
 	SetTimeIfNotNill(&cluster.Status.LastUpdateTime, c.UpdatedAt)
 	SetTimeIfNotNill(&cluster.Status.ControllerLogsCollectionTime, c.ControllerLogsCollectedAt)
 	cluster.Status.ID = c.ID.String()
+	cluster.Status.Error = ""
+}
+
+func (r *ClusterReconciler) updateState(ctx context.Context, cluster *adiiov1alpha1.Cluster, c *common.Cluster) (ctrl.Result, error) {
+	r.syncClusterState(cluster, c)
 
 	if err := r.Status().Update(ctx, cluster); err != nil {
 		r.Log.WithError(err).Error("failed to update error status")
-		return ctrl.Result{}, errors.Wrap(err, "failed to create cluster")
+		return ctrl.Result{Requeue: true}, nil
 	}
 
-	return ctrl.Result{}, nil
+	r.Log.Infof("Updated cluster %s %s state", cluster.Name, cluster.Namespace)
+	return ctrl.Result{RequeueAfter: defaultRequeueClusterDuration}, nil
```

Comment:
not sure I understand why do we set `RequeueAfter` on success, what am I missing?

Reply:
It's a temporary solution until we have a support for backend notifications from the monitors, lets stay user added hosts to the cluster and now in the backend it will become known, how will the resource be updated? so for now i just reconcile again and take the new state from the backend every few seconds.

Reply:
for now solved it with https://github.com/openshift/assisted-service/pull/891#issuecomment-757694849